pkg/apiserver/apiserver.go

```diff
@@ -241,12 +244,48 @@ func (s *APIServer) Router() (*gin.Engine, error) {
 	return s.router, nil
 }
 
+func (s *APIServer) GetTLSConfig() (*tls.Config, error) {
+	var caCert []byte
+	var err error
+	var caCertPool *x509.CertPool
+	var clientAuthType = tls.VerifyClientCertIfGiven //tls.ClientAuthType(s.TLS.ClientVerification)
+
+	if s.TLS == nil {
+		return &tls.Config{}, nil
+	}
+
+	if clientAuthType > tls.RequestClientCert {
+		caCert, err = ioutil.ReadFile(s.TLS.CACertPath)
+		if err != nil {
+			return nil, errors.Wrap(err, "Error opening cert file")
+		}
+		caCertPool = x509.NewCertPool()
+		caCertPool.AppendCertsFromPEM(caCert)
+	}
+
+	return &tls.Config{
+		ServerName: s.TLS.ServerName, //should it be removed ?
+		// ClientAuth: tls.NoClientCert,				// Client certificate will not be requested and it is not required
+		// ClientAuth: tls.RequestClientCert,			// Client certificate will be requested, but it is not required
+		// ClientAuth: tls.RequireAnyClientCert,		// Client certificate is required, but any client certificate is acceptable
+		// ClientAuth: tls.VerifyClientCertIfGiven,		// Client certificate will be requested and if present must be in the server's Certificate Pool
+		// ClientAuth: tls.RequireAndVerifyClientCert,	// Client certificate will be required and must be present in the server's Certificate Pool
+		ClientAuth: clientAuthType, //tls.ClientAuthType() tls.VerifyClientCertIfGiven,
```

Comment:
maybe we can find something more friendly than the default enum (currently a raw mapping of `tls.VerifyClientCertIfGiven`, `tls.RequireAnyClientCert` etc.)